windows-agent/internal/config/config.go

```diff
@@ -32,6 +32,9 @@ type Config struct {
 
 	// Sync
 	mu *sync.Mutex
+
+	// notifiers are called after any configuration changes.
+	notifiers []func()
```

Comment:
I'd have named this as `listeners` or `observers` since they receive a notification (qualifying the Config as the notifier - who emits the notifications).